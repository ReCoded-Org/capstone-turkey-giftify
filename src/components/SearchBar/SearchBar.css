.itemImage {
  height: 150px;
}

.cards {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.oneCard {
  width: 20%;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  margin: 5%;
}

.oneCard:hover {
  box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
}

.form {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
}

.searchBar {
  height: 50px;
  width: 300px;
  border-radius: 20px;
}

.Search {
  height: 30px;
  width: 100px;
  border-radius: 13px;
  margin-top: 20px;
  background-color: #023e8a;;
  color: white;
  font-style: normal;
}

.parag {
  color: white;
  text-align: center;
  margin-bottom: 5%;
}
