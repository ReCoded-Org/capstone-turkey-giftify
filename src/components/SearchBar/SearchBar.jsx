import React, { useState, useEffect } from "react";
import "./SearchBar.css";
import { db } from "../../firebase";
import Boards from "../boards/Boards";

function SearchSection() {
  const [allItems, setAllItems] = useState([]);
  const [input, setInput] = useState("");
  const [testSearchValue, setTestSearchValue] = useState("");

  useEffect(() => {
    async function fetchData() {
      try {
        db.collection("categories")
          .get()
          .then((snapshot) => {
            snapshot.forEach((doc) => {
              let docRef = db
                .collection("categories")
                .doc(doc.id)
                .collection("items");
              docRef.get().then((querySnapshot) => {
                querySnapshot.forEach((item) => {
                  setAllItems((prevState) => {
                    return { ...prevState, [doc.id]: [item.data()] };
                  });
                });
              });
            });
          });
      } catch (err) {
        alert(err);
      }
    }
    fetchData();
  }, []);

  function handleChange(event) {
    setInput(event.target.value);
  }

  function handleSubmit(event) {
    event.preventDefault();
    setTestSearchValue(input);
  }

  return (
    <div>
      <h2 className="parag">Or would you like to search for yourself ?</h2>

      <form className="form" onSubmit={handleSubmit}>
        <label>
          <input
            className="searchBar"
            type="text"
            name="name"
            onChange={handleChange}
            value={input}
          />
        </label>
        <input className="submit" type="submit" value="Submit" />
      </form>

      <Boards
        allItems={allItems}
        input={input}
        testSearchValue={testSearchValue}
      />
    </div>
  );
}

export default SearchSection;
