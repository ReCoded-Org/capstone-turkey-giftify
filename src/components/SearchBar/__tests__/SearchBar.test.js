import React from "react";
import SearchBar from "./../SearchBar";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("SearchBar rendered correctly", () => {
  const tree = renderer.create(<SearchBar />).toJSON();
  expect(tree).toMatchSnapshot();
});
