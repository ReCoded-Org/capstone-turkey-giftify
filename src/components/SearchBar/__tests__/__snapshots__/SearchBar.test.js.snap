// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Received 1`] = `
<div>
  <h2
    className="parag"
  >
    Or would you like to search for yourself ?
  </h2>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <label>
      <input
        className="searchBar"
        name="name"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <input
      className="submit"
      type="submit"
      value="Submit"
    />
  </form>
  <div>
    <div />
  </div>
</div>
`;
